lib/bosh/BoshDirectorClient.js

```diff
@@ -16,27 +16,133 @@ const NotFound = errors.NotFound;
 const BadRequest = errors.BadRequest;
 
 class BoshDirectorClient extends HttpClient {
-  constructor(boshDirectorConfig) {
-    const boshDirector = boshDirectorConfig || config.director;
+  constructor() {
+    // activeConfigs - directors supporting lifecycle oprations
+    let activeConfigs = BoshDirectorClient.getActiveConfigs();
+    // directorConfig - directors supporting 'create' oprations
+    let coreConfigs = BoshDirectorClient.getCoreConfigs(activeConfigs);
+    // inactiveConfigs - directors for OutOfBand backup scenario
+    let inactiveConfigs = BoshDirectorClient.getInactiveConfigs();
```

Comment:
Naming of various configs not intuitive. 
1. Usage of inactive is incorrect, they are active but for the scope of SF they are to be treated only for OOB backup/restore. So can either call them as either 'secondaryConfigs' or 'oobDirectorConfigs' or anyother name whichever u seem is fit.
2. activeConfig - Comment reads lifecycle operations but 'create' is also a LC operation. Can name them as just 'Primary'.
3. coreConfigs - does not indicate what is meant by core. Can rather name them as 'activePrimary'. This config is nothing but primary configs, which support create as well.